File: services/internal-treasury/src/main/java/br/com/deltaglobalbank/internal_treasury/infrastructure/persistence/mappers/MakePixMapper.java
package br.com.deltaglobalbank.internal_treasury.infrastructure.persistence.mappers;

import br.com.deltaglobalbank.internal_treasury.domain.makePix.MakePix;
import br.com.deltaglobalbank.internal_treasury.domain.shared.PaymentsStatus;
import br.com.deltaglobalbank.internal_treasury.infrastructure.persistence.entities.MakePixEntity;

public final class MakePixMapper {

    private MakePixMapper() {
    }

    public static MakePix toDomain(MakePixEntity entity) {
        return new MakePix(
            entity.getId(),
            entity.getAccountId(),
            entity.getRecipientInstitutionCode(),
            entity.getRecipientBranchCode(),
            entity.getRecipientAccountNumber(),
            PaymentsStatus.valueOf(entity.getStatus()),
            entity.getPaidAt(),
            entity.getRecipientAccountType(),
            entity.getRecipientName(),
            entity.getOperationAmount(),
            entity.getCreatedAt()
        );
    }

    public static MakePixEntity toEntity(MakePix domain) {
        return new MakePixEntity(
            domain.id(),
            domain.accountId(),
            domain.recipientInstitutionCode(),
            domain.recipientBranchCode(),
            domain.recipientAccountNumber(),
            domain.recipientAccountType(),
            domain.recipientName(),
            domain.operationAmount(),
            domain.paidAt(),
            domain.createdAt(),
            domain.status().name()
        );
    }
}

File: services/internal-treasury/src/main/kotlin/br/com/deltaglobalbank/internal_treasury/infrastructure/persistence/mappers/MakePixMapper.kt
package br.com.deltaglobalbank.internal_treasury.infrastructure.persistence.mappers

import br.com.deltaglobalbank.internal_treasury.domain.makePix.MakePix
import br.com.deltaglobalbank.internal_treasury.domain.shared.PaymentsStatus
import br.com.deltaglobalbank.internal_treasury.infrastructure.persistence.entities.MakePixEntity

fun MakePixEntity.toDomain(): MakePix = MakePix(
    id = this.id,
    accountId = this.accountId,
    recipientInstitutionCode = this.recipientInstitutionCode,
    recipientBranchCode = this.recipientBranchCode,
    recipientAccountNumber = this.recipientAccountNumber,
    recipientAccountType = this.recipientAccountType,
    recipientName = this.recipientName,
    paidAt = this.paidAt,
    operationAmount = this.operationAmount,
    createdAt = this.createdAt,
    status = PaymentsStatus.valueOf(this.status),
)

fun MakePix.toEntity(): MakePixEntity = MakePixEntity(
    id = this.id,
    accountId = this.accountId,
    recipientInstitutionCode = this.recipientInstitutionCode,
    recipientBranchCode = this.recipientBranchCode,
    recipientAccountNumber = this.recipientAccountNumber,
    recipientAccountType = this.recipientAccountType,
    paidAt = this.paidAt,
    recipientName = this.recipientName,
    operationAmount = this.operationAmount,
    createdAt = this.createdAt,
    status = this.status.name,
)